homeassistant/components/aidot/coordinator.py

```diff
@@ -0,0 +1,158 @@
+
+from homeassistant.components.sensor import timedelta
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+type AidotConfigEntry = ConfigEntry[AidotDeviceManagerCoordinator]
+_LOGGER = logging.getLogger(__name__)
+
+
+class AidotDeviceUpdateCoordinator(DataUpdateCoordinator[DeviceStatusData]):
+    """Class to manage Aidot data."""
+
+    device: dict[str, Any]
+    ip_address: str
+    device_client: DeviceClient
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: AidotConfigEntry,
+        device_client: DeviceClient,
+    ) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=config_entry,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=1),
+        )
+        self.device_client = device_client
+        self.identifier = config_entry.entry_id
+
+    async def _async_setup(self) -> None:
+        """Set up the coordinator.
+
+        Can be overwritten by integrations to load data or resources
+        only once during the first refresh.
+        """
+        try:
+            await self.device_client.async_login()
+        except AidotUserOrPassIncorrect as error:
+            raise ConfigEntryError from error
+
+    async def _async_update_data(self) -> DeviceStatusData:
+        """Update data async."""
+        try:
+            if self.device_client.connect_and_login is False:
+                await self.device_client.async_login()
+            status = await self.device_client.read_status()
+        except AidotNotLogin:
+            status = self.device_client.status
+            status.online = False
+        return status
+
+
+class AidotDeviceManagerCoordinator(DataUpdateCoordinator[None]):
+    """Class to manage fetching Aidot data."""
+
+    config_entry: AidotConfigEntry
+    client: AidotClient
+    device_coordinators: dict[str, AidotDeviceUpdateCoordinator]
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: AidotConfigEntry,
+    ) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=config_entry,
+            name=DOMAIN,
+            update_interval=timedelta(hours=6),
+        )
+        self.client = AidotClient(
+            session=async_get_clientsession(hass),
+            token=config_entry.data[CONF_LOGIN_INFO],
+        )
+        self.client.start_discover()
+        self.client.set_token_fresh_cb(self.token_fresh_cb)
+        self.identifier = config_entry.entry_id
+        self.device_coordinators = {}
```

Comment:
```suggestion

    def __init__(
        self,
        hass: HomeAssistant,
        config_entry: AidotConfigEntry,
    ) -> None:
        """Initialize coordinator."""
        super().__init__(
            hass,
            _LOGGER,
            config_entry=config_entry,
            name=DOMAIN,
            update_interval=timedelta(hours=6),
        )
        self.client = AidotClient(
            session=async_get_clientsession(hass),
            token=config_entry.data[CONF_LOGIN_INFO],
        )
        self.client.start_discover()
        self.client.set_token_fresh_cb(self.token_fresh_cb)
        self.identifier = config_entry.entry_id
        self.device_coordinators: dict[str, AidotDeviceUpdateCoordinator] = {}
```
that might be preference, but I think it's clearer that way, if you just type the device coordinators in here